packages/plugins/i18n/admin/src/components/LocaleSelect/index.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Select, Option } from '@strapi/parts/Select';
+import { Loader } from '@strapi/parts/Loader';
+import { useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import useLocales from '../../hooks/useLocales';
+import useDefaultLocales from '../../hooks/useDefaultLocales';
+import { getTrad } from '../../utils';
+
+const SmallLoader = styled(Loader)`
+  img {
+    height: 1rem;
+    width: 1rem;
+  }
+`;
+
+/**
+ * The component is memoized and needs a useCallback over the onLocaleChange and
+ * onClear props to prevent the Select from re-rendering N times when typing on a specific
+ * key in a formik form
+ */
+const LocaleSelect = React.memo(({ value, onLocaleChange, error, onClear }) => {
+  const { formatMessage } = useIntl();
+  const { defaultLocales, isLoading } = useDefaultLocales();
+  const { locales } = useLocales();
+
+  const options = (defaultLocales || [])
+    .map(locale => ({
+      label: locale.name,
+      value: locale.code,
+    }))
+    .filter(({ value: v }) => {
+      const foundLocale = locales.find(({ code }) => code === v);
+
+      return !foundLocale;
+    });
+
+  const computedValue = value || '';
```

Comment:
`const computedValue = value || options[0]?.value || '';`

Reply:
it won't match the default display name if we do that